From 348ca7e5e83bc36101172993f3aa60ae8ae5a2c7 Mon Sep 17 00:00:00 2001
From: Peter Jung <admin@ptr1337.dev>
Date: Fri, 18 Oct 2024 22:44:33 +0200
Subject: [PATCH 6/6] silence-event-assert-until-570

Currently, when playing CS2 it reports a massive spam in the dmesg log. This will be fixed in the 570 driver.
Silence it for now.

Signed-off-by: Peter Jung <admin@ptr1337.dev>
---
 src/nvidia/src/kernel/rmapi/event_notification.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/nvidia/src/kernel/rmapi/event_notification.c b/src/nvidia/src/kernel/rmapi/event_notification.c
index cf78eadd..d6937cac 100644
--- a/src/nvidia/src/kernel/rmapi/event_notification.c
+++ b/src/nvidia/src/kernel/rmapi/event_notification.c
@@ -286,11 +286,11 @@ static NV_STATUS _gpuEngineEventNotificationListNotify
     portSyncSpinlockAcquire(pEventNotificationList->pSpinlock);
     {
         // We don't expect this to be called multiple times in parallel
-        NV_ASSERT_OR_ELSE(pEventNotificationList->pendingEventNotifyCount == 0,
+        if (pEventNotificationList->pendingEventNotifyCount != 0)
         {
             portSyncSpinlockRelease(pEventNotificationList->pSpinlock);
             return NV_ERR_INVALID_STATE;
-        });
+        }
 
         EngineEventNotificationListIter it =
             listIterAll(&pEventNotificationList->eventNotificationList);
-- 
2.47.0

